Automattic/wp-calypso
# Use `ThemeTypeBadge` in theme showcase

Related to https://github.com/Automattic/dotcom-forge/issues/2438

## Proposed Changes

Refactors the Theme Showcase to use the `ThemeTypeBadge` component introduced in https://github.com/Automattic/wp-calypso/pull/77554 to display the theme type badges.

The Design Picker will be refactored in a follow-up PR.

**Free themes**
<img width="96" alt="Screenshot 2023-05-30 at 17 52 35" src="https://github.com/Automattic/wp-calypso/assets/1233880/71f145f4-c444-47dd-a573-29e2449bb6cc">


**Premium themes**

w/o eligible plan | w/ eligible plan | w/ one-off purchase
--- | --- | ---
<img width="321" alt="Screenshot 2023-05-30 at 17 54 19" src="https://github.com/Automattic/wp-calypso/assets/1233880/e871c198-3c73-42a2-bef8-4312b8905d1c"> | <img width="311" alt="Screenshot 2023-05-31 at 12 16 20" src="https://github.com/Automattic/wp-calypso/assets/1233880/5613b603-c374-4f72-93e3-c9f00bab06ab"> | <img width="271" alt="Screenshot 2023-05-31 at 12 18 03" src="https://github.com/Automattic/wp-calypso/assets/1233880/8c5af615-79c3-4e12-bcc9-2c63dcc13ddb">


**dotOrg themes**

w/o eligible plan | w/ eligible plan
--- | ---
<img width="313" alt="Screenshot 2023-05-31 at 12 18 41" src="https://github.com/Automattic/wp-calypso/assets/1233880/1550a6ca-e6f6-4584-9d27-056b9fc94000"> | <img width="314" alt="Screenshot 2023-05-31 at 12 19 05" src="https://github.com/Automattic/wp-calypso/assets/1233880/e55f51de-4745-4721-b6ad-0dc06a0ad1a4">

**WooCommerce themes**

w/o eligible plan | w/ eligible plan
--- | --- 
<img width="316" alt="Screenshot 2023-05-31 at 12 19 34" src="https://github.com/Automattic/wp-calypso/assets/1233880/51c34cd0-bffe-492f-ace8-03ede29d0701"> | <img width="311" alt="Screenshot 2023-05-31 at 12 20 02" src="https://github.com/Automattic/wp-calypso/assets/1233880/3aceda01-729f-46a8-bc28-2a8b3b9382b9">

**Marketplace themes**

w/o plan & w/o subscription | w/ plan & w/o subscription | w/o plan & w/ subscription | w/ plan & subscription
--- | --- | --- | ---
<img width="315" alt="Screenshot 2023-05-31 at 12 20 43" src="https://github.com/Automattic/wp-calypso/assets/1233880/6179771d-c199-4904-af84-079565fa3edc"> | <img width="307" alt="Screenshot 2023-05-31 at 12 21 21" src="https://github.com/Automattic/wp-calypso/assets/1233880/94d1ae01-8cca-4345-aea7-0c274ac1af37"> | <img width="313" alt="Screenshot 2023-05-31 at 12 23 29" src="https://github.com/Automattic/wp-calypso/assets/1233880/b2a67abf-a92a-474a-870e-7b4c7bd460d3"> | <img width="320" alt="Screenshot 2023-05-31 at 12 22 37" src="https://github.com/Automattic/wp-calypso/assets/1233880/38c44dc8-30d7-4673-881c-0c8f97109512">


## Testing Instructions

- Use the Calypso live link below.
- Go to Appearance > Themes.
- Test all the different scenarios listed in the screenshots above.
- Make sure the theme type badge is rendered correctly.
